cloudpathlib/cloudpath.py

```diff
@@ -594,14 +594,15 @@ def download_to(self, destination: Union[str, os.PathLike]) -> Path:
                 destination = destination / self.name
             return self.client._download_file(self, destination)
         else:
-            destination.mkdir(exist_ok=True)
+            destination.mkdir(parents=True, exist_ok=True)
```

Comment:
I'm not totally bought in on automatically creating missing parents by default. I believe most things (unix programs and Python libraries) do not normally do this. 

A related consideration here is to maybe add a `parents` parameter to `download_to`?